guides/ci/travis.md

```diff
@@ -0,0 +1,358 @@
+
+addons:
+  apt:
+    update: true
+
+script:
+  - sqlite3 --version
+  - crystal spec
+```
+
+**Note:** We are using `sqlite3` just for printing SQLite's version, as an example of external dependency.
+
+The same can be accomplished using a containerization service like Docker. Here is an example of a `Dockerfile` installing SQLite3:
+
+```dockerfile
+# Dockerfile
+FROM crystallang/crystal:latest
+
+RUN apt-get update && apt-get install -y sqlite3
+```
+
+And here is the TravisCI configuration file:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+before_install:
+  # build image using Dockerfile:
+  - docker build -t testing .
+
+script:
+  # run specs in the container
+  - docker run -v $PWD:/src -w /src testing crystal spec
+  # sqlite3
+  - docker run -v $PWD:/src -w /src testing sqlite3 --version
+```
+
+Again, these are only examples of two ways of running our application and installing (and using) dependencies.
+
+> *Note:* Dockerfile arguments can be used to use the same Dockerfile for latest, nightly or a specific version.
+
+## Using a database (example using MySQL)
+
+As you may notice, in the last configuration file we are using `docker` as a `service`. Travis CI may start different services as needed. For example, we may need [MySQL](https://docs.travis-ci.com/user/database-setup/#mysql).
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+services:
+  - mysql
+
+script:
+  - crystal spec
+```
+
+To continue this example, let's add a new test that uses the database.
+(**important:** the new test case is only for testing Travis CI database service, and it's not a good design or test case by any means)
+
+```crystal
+# game_of_life_spec.cr
+require "./spec_helper"
+
+describe "a new world" do
+  it "should be empty" do
+    world = World.new
+    world.is_empty?.should be_true
+  end
+end
+
+describe "an empty world" do
+  it "should not be empty after adding a cell" do
+    world = World.empty
+    world.set_living_at(Location.random)
+    world.is_empty?.should be_false
+  end
+
+  it "should be persisted" do
+    World.empty.save
+  end
+end
+```
+
+And the new implementation:
+
+```crystal
+# game_of_life.cr
+require "mysql"
+
+class Location
+  getter x : Int32
+  getter y : Int32
+
+  def self.random
+    Location.new(Random.rand(10), Random.rand(10))
+  end
+
+  def initialize(@x, @y)
+  end
+end
+
+class World
+  @living_cells : Array(Location)
+
+  def self.empty
+    World.new([] of Location)
+  end
+
+  def initialize(living_cells = [] of Location)
+    @living_cells = living_cells
+  end
+
+  def set_living_at(a_location)
+    @living_cells << a_location
+  end
+
+  def is_empty?
+    @living_cells.size == 0
+  end
+
+  def save
+    DB.connect ENV["DATABASE_URL"] do |cnn|
+      cnn.exec("insert into worlds (living_cells) values (?);", @living_cells.size)
+    end
+  end
+end
+```
+
+We may use this `docker-compose` configuration to have a local MySQL database for our test to run:
```

Comment:
This adds even more complexity and it doesn't even explain why you would need docker compose when you can install a mysql service in travis.

Reply:
The locally running examples will go away and reduce the length.